### PR TITLE
added support for custom paths in release-to-candidate CI

### PR DESCRIPTION
**Pull request by soumyaDghosh**

### CODE REVIEW

File: release-to-candidate/action.yaml [hunk 1]
```diff
@@ -20,6 +20,12 @@ inputs:
   store-token:
     description: "A token with permissions to upload to the specified channel"
     required: true
+  yaml-path:
+    description: "Custom path to snapcraft.yaml allowing to have multiple manifests in one repo."
+    required: false
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:25]: See my comments on #7 - same applies here

**Review by jnsgruk — CHANGES_REQUESTED** [review]

File: release-to-candidate/action.yaml [hunk 2]
```diff
@@ -20,6 +20,12 @@ inputs:
...
+  name:
+    description: "Name of the snap"
+    required: false
```
[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply release-to-candidate/action.yaml:28]: This is also needed for #4

[the remark this answers is not in the document]
Comment (jnsgruk) [reply release-to-candidate/action.yaml:28]: In any case, can we please call this `snap-name` to make it a little more clear.

[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply release-to-candidate/action.yaml:28]: Umm.. yeah sure

File: release-to-candidate/action.yaml [hunk 3]
```diff
@@ -46,21 +52,52 @@ runs:
...
+
+
     - name: Build the snap (${{ inputs.architecture }})
       id: build
       shell: bash
+      with:
+        yaml_path: ${{ steps.yaml-path.outputs.yaml_path }}
       env:
-        name: ${{ github.event.repository.name }}
+        name: ${{ if not(inputs.name) }}${{ github.event.repository.name }}${{ else }}${{ inputs.name }}${{ endif }}
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:82]:
```suggestion
        name: ${{ inputs.name || github.event.repository.name }} 
```

I think this should work and is quite a lot more succinct

File: release-to-candidate/README.md [hunk 4]
```diff
@@ -30,6 +30,7 @@ jobs:
 | `channel`         | The channel to release the snap to.                                                       |    N     | `candidate` |
 | `launchpad-token` | A token with permissions to create Launchpad remote builds.                               |    Y     |             |
 | `store-token`     | A token with permissions to upload and release to the specified channel in the Snap Store |    Y     |             |
+| `yaml_path`       | The path to the YAML file.                                                                |    N     |`Default Paths`|
```
[the remark this answers is not in the document]
Comment (soumyaDghosh) [reply release-to-candidate/README.md]: Sorry about that. Fixed it

File: release-to-candidate/README.md [hunk 5]
```diff
@@ -24,12 +24,14 @@ jobs:
...
-| `store-token`     | A token with permissions to upload and release to the specified channel in the Snap Store |    Y     |             |
+| Key               | Description                                                                               | Required | Default        |
+| ----------------- | ----------------------------------------------------------------------------------------- | :------: | :--------------|
+| `architecture`    | The architecture for which to build the snap.                                             |    N     | `amd64`        |
+| `channel`         | The channel to release the snap to.                                                       |    N     | `candidate`    |
+| `launchpad-token` | A token with permissions to create Launchpad remote builds.                               |    Y     |                |
+| `store-token`     | A token with permissions to upload and release to the specified channel in the Snap Store |    Y     |                |
+| `yaml-path`       | The path to the YAML file.                                                                |    N     |`Default Path`  |
```
Comment (jnsgruk) [inline release-to-candidate/README.md:33]:
```suggestion
| `snapcraft-yaml-path`       | The path to the Snapcraft YAML file.                                                                |    N     |    |
```

File: release-to-candidate/README.md [hunk 6]
```diff
@@ -24,12 +24,14 @@ jobs:
...
+| `snap-name`       | The name of the snap.                                                                     |    N     |`Repo Name`     |
```
Comment (jnsgruk) [inline release-to-candidate/README.md:34]:
```suggestion
| `snap-name`       | The name of the snap.                                                                     |    N     |          |
```

File: release-to-candidate/action.yaml [hunk 7]
```diff
@@ -69,12 +106,14 @@ runs:
...
       shell: bash
+      with:
+        yaml_path: ${{ steps.yaml-path.outputs.yaml_path }}
       run : |
         # Populate defaults
         echo "classic=false" >> "$GITHUB_OUTPUT"
 
         # Check for classic confinement and update the output if the snap is classic
-        if [[ "$(cat snap/snapcraft.yaml | yq -r '.confinement')" == "classic" ]]; then
+        if [[ "$(cat ${yaml_path} | yq -r '.confinement')" == "classic" ]]; then
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:116]: This same logic will need to be ported to the `test-snap-can-build` workflow

Comment (soumyaDghosh) [reply release-to-candidate/action.yaml:116]: Yes, working on all of them one by one

File: release-to-candidate/README.md (hunk 5 above)
Comment (jnsgruk) [reply release-to-candidate/README.md:33]: The default is computed if the input is left empty, but there is no default value set in the input specification itself.

Comment (soumyaDghosh) [reply release-to-candidate/README.md:33]: Oooo.. I thought these paths are default in snapcraft. Thanks for the info

File: release-to-candidate/action.yaml [hunk 8]
```diff
@@ -43,23 +49,56 @@ runs:
...
-        cat snap/snapcraft.yaml | yq 'del(.architectures)' | sponge snap/snapcraft.yaml
+        yq -i 'del(.architectures)' "$yaml_path"
 
         snapcraft remote-build --launchpad-accept-public-upload --build-for="${arch}"
 
-        version="$(cat snap/snapcraft.yaml | yq -r '.version')"
+        version="$(yq -r '.version' "$yaml_path")"
+        if [[ ! -e ${name}_${version}_${arch}.snap ]]; then
+            echo "Could not find ${name}_${version}_${arch}.snap"
+            cat ${name}_${arch}.txt
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:97]: What is this for? Either way it needs to be quoted :)

File: release-to-candidate/action.yaml [hunk 9]
```diff
@@ -43,23 +49,56 @@ runs:
...
+            exit 1
+        else
+            cat ${name}_${arch}.txt
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:100]: Same here? Why are we just `cat`ing this?

File: release-to-candidate/action.yaml (hunk 8 above)
Comment (soumyaDghosh) [reply release-to-candidate/action.yaml:97]: It's the build log, #3

File: release-to-candidate/action.yaml [hunk 10]
```diff
@@ -94,10 +94,10 @@ runs:
         version="$(yq -r '.version' "$yaml_path")"
         if [[ ! -e ${name}_${version}_${arch}.snap ]]; then
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:95]:
```suggestion
        if [[ ! -e "${name}_${version}_${arch}.snap" ]]; then
```

**Review by jnsgruk — DISMISSED** [review]
LGTM, thanks for the iteration.

File: release-to-candidate/action.yaml [hunk 11]
```diff
@@ -43,23 +47,57 @@ runs:
...
         git config --global user.name "Github Actions"
 
-        # Install moreutils so we have access to sponge
-        sudo apt-get update; sudo apt-get install -y moreutils
+    - name: Find the snapcraft.yaml path
+      id: yaml-path
+      shell: bash
+      run : |
+        if [[ -n "${{ inputs.snapcraft-yaml-path }}"; then
+          yaml-path="${{ inputs.snapcraft-yaml-path }}"
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:55]: Sorry, I appreciate I might have been confusing in a past comment, but you can't name `bash` variables with `-`.

Your Github inputs/outputs can use hyphens, but `bash` variable names need to use `_` or it's a syntax error

File: release-to-candidate/action.yaml [hunk 12]
```diff
@@ -43,23 +47,57 @@ runs:
...
+        fi
+        echo "yaml-path=${yaml-path}" >> "$GITHUB_OUTPUT"
+        echo "snap-name=$(yq -r '.name' "$yaml-path")" >> "$GITHUB_OUTPUT"
+
 
     - name: Build the snap (${{ inputs.architecture }})
       id: build
       shell: bash
+      with:
+        yaml-path: ${{ steps.yaml-path.outputs.yaml-path }}
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:82]: This isn't correct syntax - you need to remove the `with` here, and create a `yaml_path` entry in the `env` map

File: release-to-candidate/action.yaml [hunk 13]
```diff
@@ -69,12 +107,14 @@ runs:
     - name: Parse snap review information
       id: parse
       shell: bash
+      with:
+        yaml-path: ${{ steps.yaml-path.outputs.yaml-path }}
```
Comment (jnsgruk) [inline release-to-candidate/action.yaml:111]: This needs to be in `env`, not `with`

File: release-to-candidate/action.yaml (hunk 11 above)
Comment (soumyaDghosh) [reply release-to-candidate/action.yaml]: Check if it's fine now..

**Review by jnsgruk — APPROVED** [review]
LGTM, thanks

**Pull request merged**